### PR TITLE
Add Logisim

### PR DESCRIPTION
**Pull request by turtlebasket**
## Description
Adding support for [Logisim (Evolution)](https://github.com/logisim-evolution/logisim-evolution) files. Logisim is a popular FOSS tool for digital system design, typically used for educational purposes. Logisim files are actually just XML but contain circuit logic that can be viewed and simulated in Logisim, making the `.circ` format somewhat comparable to `.ipynb`.

## Checklist:

- [ ] **I am adding a new extension to a language.**
  - [ ] The new extension is used in hundreds of repositories on GitHub.com
    - Search results for each extension:
      
      - https://github.com/search?q=path%3A*.circ+lang%3Axml&type=code
  - [ ] I have included a real-world usage sample for all extensions added in this PR:
    - Sample source(s):
      - [URL to each sample source, if applicable]
    - Sample license(s):
  - [ ] I have included a change to the heuristics to distinguish my language from others using the same extension.

- [x] **I am adding a new language.**
  - [x] The extension of the new language is used in hundreds of repositories on GitHub.com.
    - Search results for each extension:
      
      -  https://github.com/search?type=code&q=NOT+is%3Afork+path%3A*.circ+language%3Axml
  - [x] I have included a real-world usage sample for all extensions added in this PR:
    - Sample source(s):
      - https://github.com/sam-astro/Astro8-Computer/blob/main/cpu-circuit.circ
      - https://github.com/rosco-m68k/rosco_m68k/blob/develop/design/r0/logisim/address_decode_v2.circ
    - Sample license(s): MIT
  - [ ] I have included a syntax highlighting grammar: [URL to grammar repo]
      
  - [x] I have added a color
    - Hex value: `#008016`
    - Rationale: Just picked the shade of green from the Logisim logo
  - [x] I have updated the heuristics to distinguish my language from others using the same extension.

- [ ] **I am fixing a misclassified language**
  - [ ] I have included a new sample for the misclassified language:
    - Sample source(s):
      - [URL to each sample source, if applicable]
    - Sample license(s):
  - [ ] I have included a change to the heuristics to distinguish my language from others using the same extension.

- [ ] **I am changing the source of a syntax highlighting grammar**
  - Old: [URL to grammar repo]
  - New: [URL to grammar repo]

- [ ] **I am updating a grammar submodule**
  

- [ ] **I am adding new or changing current functionality**
  
  - [ ] I have added or updated the tests for the new or changed functionality.

- [ ] **I am changing the color associated with a language**
  
  - [ ] I have obtained agreement from the wider language community on this color change.
    - [URL to public discussion]
    - [Optional: URL to official branding guidelines for the language]

### CODE REVIEW

**Review by lildude — CHANGES_REQUESTED** [review]
As this is standard XML, the extension should be added to the XML language entry. We try to limit the splitting of things like XML and JSON to new entries only when they deviate and have their own grammar.

The `samples/Logisim/astro8_cpu.circ` sample is also way too big. Please remove it or replace it with a smaller sample.

**Comment by turtlebasket** [comment]
Sounds like it'd be appropriate to convert this PR to add the `.circ` extension to XML and move the samples (except for the large one) to `samples/XML` instead?

**Comment by turtlebasket** [comment]
Alrighty, updated- lmk if anything looks off with this

File: samples/XML/address_decode_v2.circ [hunk 1]
```diff
@@ -0,0 +1,353 @@
+<?xml version="1.0" encoding="UTF-8" standalone="no"?>
```
Comment (Alhadis) [inline samples/XML/address_decode_v2.circ:1]: @turtlebasket Are there many `.circ` files that lack an XML header? If not, then these files should be classified correctly as XML on GitHub, as Linguist has a [dedicated strategy](https://github.com/github-linguist/linguist/blob/ff004eaeb54e868e3b1707f0584ceb3684b155f2/lib/linguist/strategy/xml.rb) for recognising files that begin with the line

```xml
<?xml version=
```

If many `.circ` files in-the-wild lack these headers and aren't receiving any highlighting, then this addition is still valid (and you can dismiss this feedback).

**Review by julitabarelkowska — APPROVED** [review]

**Comment by lildude** [comment]
The search provided in the body of this PR includes the `language=xml` qualifier which suggests these files are already being successfully detected as XML. Negating that clearly returns files which are not XML. Merging this PR would change this and mark them incorrectly as XML.

As such, this PR appears to be completely unnecessary, and would in fact cause incorrect classification and not correct any.

Closing as this is not needed.

**Pull request closed without merging**